hrm-service/src/contact-job/contact-job-complete.ts

```diff
@@ -44,18 +48,34 @@ export const pollAndprocessCompletedContactJobs = async () => {
       try {
         const completedJob: CompletedContactJobBody = JSON.parse(m.Body);
 
-        await processCompletedContactJob(completedJob);
+        if (completedJob.attemptResult === 'success') {
+          await processCompletedContactJob(completedJob);
+
+          // Mark the job as completed
+          const markedComplete = await completeContactJob(
+            completedJob.jobId,
+            completedJob.attemptPayload,
+          );
+
+          // Delete the message from the queue (this could be batched)
+          await deletedCompletedContactJobs(m.ReceiptHandle);
+
+          return markedComplete;
+        } else {
+          const { jobId, attemptNumber, attemptPayload } = completedJob;
```

Comment:
Do we not need to call `deletedCompletedContactJobs` on a failed job?

Reply:
Absolutely. Thanks for the catch!